Assets/Mirage/ObjectPolling/ObjectPoolManager.cs

```diff
@@ -0,0 +1,204 @@
+        private void UnSpawnObject<T>(T spawned) where T : Object
+        {
+            ObjectPooling<T> p = default;
+
+            foreach (object pool in _recycledObjects.Values)
+            {
+                if (!(pool is ObjectPooling<T> obj) || !obj.PooledObjectIDs.Contains(spawned.GetInstanceID())) continue;
+
+                p = obj;
+
+                break;
+            }
+
+            p.Despawn(spawned);
+        }
+
+        #endregion
+
+        #region Network Spawning
+
+        /// <summary>
+        ///     Spawn an object using our pooled objects for the specific object
+        ///     or create a new pool if no pool has been created yet.
+        /// </summary>
+        /// <param name="position">The position we want to spawn the object at.</param>
+        /// <param name="rotation"></param>
+        /// <param name="AssetId">The asset id of the object from <see cref="NetworkIdentity" /> to use to spawn object of.</param>
+        /// <param name="quantity">How many pooled objects we want to create of this object. If we don't have a pool started.</param>
+        /// <returns></returns>
+        internal NetworkIdentity NetworkSpawnPool(Vector3 position, Quaternion rotation, Guid AssetId, int quantity = 3)
+        {
+            int prefabId = _objectsAssetIds[AssetId].gameObject.GetInstanceID();
+
+            if (_recycledObjects.ContainsKey(prefabId))
+                return (_recycledObjects[prefabId] is ObjectPooling<NetworkIdentity>
+                    ? (ObjectPooling<NetworkIdentity>)_recycledObjects[prefabId]
+                    : default).NetworkSpawn(position, rotation);
+
+            var createParent = new GameObject(_objectsAssetIds[AssetId].name) { transform = { parent = _parent.transform } };
+
+            _recycledObjects[prefabId] = new ObjectPooling<NetworkIdentity>(_objectsAssetIds[AssetId], quantity, createParent);
+
+            return (_recycledObjects[prefabId] is ObjectPooling<NetworkIdentity>
+                ? (ObjectPooling<NetworkIdentity>)_recycledObjects[prefabId]
+                : default).NetworkSpawn(position, rotation);
+        }
+
+        /// <summary>
+        ///     Network object pooler unspawning of object and resetting parent back
+        ///     to the object pooler.
+        /// </summary>
+        /// <param name="objectSpawned">The object we want to set in active and despawn.</param>
+        /// <param name="isServer">Whether or not this is server telling us to unspawn object.</param>
+        /// <param name="isEnemy">Whether or not this is an enemy being despawn-ed.</param>
+        internal void NetworkUnSpawnObject(NetworkIdentity objectSpawned, bool isServer, bool isEnemy)
+        {
+            UnSpawnObject(objectSpawned);
+
+            // If this is network server then tell rest of clients to unspawn object.
+            if (_serverObjectManager == null || !isServer) return;
```

Comment:
would be easier to understand if this `if` was flipped; `if server then call destroy`

Reply:
What does it matter if it returns or calls destroy its doing the same thing.